### PR TITLE
 Enabling vector length 8 for vectorized elementwise kernel

### PR DESCRIPTION
**Pull request by jerrymannil**
enabling use of vector length 8 on AMD GPUs for vectorized elementwise kernel and minor parameter tuning to improve performance

Co-author: @akadutta

### CODE REVIEW

**Comment by rocm-mici** [comment]
Jenkins build for [08a6c3a7fc83d464dd798bdd986c0916cb49c85c](https://github.com/ROCm/pytorch/commit/08a6c3a7fc83d464dd798bdd986c0916cb49c85c) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1620/1//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1620/1/artifact)

**Comment by rocm-mici** [comment]
Jenkins build for [08a6c3a7fc83d464dd798bdd986c0916cb49c85c](https://github.com/ROCm/pytorch/commit/08a6c3a7fc83d464dd798bdd986c0916cb49c85c) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1620/2//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1620/2/artifact)

**Comment by rocm-mici** [comment]
Jenkins build for [08a6c3a7fc83d464dd798bdd986c0916cb49c85c](https://github.com/ROCm/pytorch/commit/08a6c3a7fc83d464dd798bdd986c0916cb49c85c) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1620/3//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1620/3/artifact)

File: aten/src/ATen/native/cuda/MemoryAccess.cuh [hunk 1]
```diff
@@ -350,11 +350,23 @@ inline C10_HOST_DEVICE int can_vectorize_up_to(const char *pointer) {
...
   constexpr int vec2_alignment = std::alignment_of<aligned_vector<scalar_t, 2>>::value;
   constexpr int vec4_alignment = std::alignment_of<aligned_vector<scalar_t, 4>>::value;
-  if (address % vec4_alignment == 0) {
-    return 4;
-  } else if (address % vec2_alignment == 0) {
-    return 2;
-  }
+  #if defined(USE_ROCM)
+    constexpr int vec8_alignment = std::alignment_of<aligned_vector<scalar_t, 8>>::value;
+    constexpr bool half_dtype = std::is_same_v<c10::BFloat16, scalar_t> || std::is_same_v<c10::Half, scalar_t>;
```
Comment (jerrymannil) [inline aten/src/ATen/native/cuda/MemoryAccess.cuh:355]: Add check for half dtype

**Review by akadutta — APPROVED** [review]
looks good to me

**Pull request merged**